Initial landing and project ideas for GSoC 2022 (#100)
keptn/community: mentorship/gsoc/README.md

# Google Summer of Code in Keptn

Google Summer of Code (GSoC) is a global, online program focused on
bringing new contributors into open source software development.
GSoC contributors work with an open source organization on a
summer programming project under the guidance of mentors.

GSoC contributors accepted into the program receive a stipend, paid by Google,
to work well-defined projects to improve or enhance the Keptn project and its ecosystem.
In exchange, numerous community members volunteer as "mentors" for GSoC contributors 
to help integrate them into the open source community and succeed in completing their projects.

## GSoC 2022

- [2022 Project ideas](./2022/project-ideas.md)
- [Google Summer of Code website](https://summerofcode.withgoogle.com/)
- [GSoC Timeline](https://summerofcode.withgoogle.com/how-it-works#timeline)

## Participate

> We invite GSoC contributors to submit project proposal applications in relation to these ideas.

Reviewer: Didn't you say that we would also consider proposals that are unrelated to these ideas?  Maybe add "or other issues that interest you" or something?

Author: ```suggestion
We invite GSoC contributors to submit project proposal applications in relation to these ideas.
Original ideas will be welcome and appreciated too!
```

### Apply as a contributor

1. Check out our project ideas.
2. Join the [Keptn Slack](https://keptn.sh/community/#slack),
   and the `#gsoc` channel there.

> 3. Explore the project ideas and find one or two interesting to you.

Reviewer: ```suggestion
3. Explore the project ideas and find one or two that interest you.
```

   Notify us in the chat.

> 4. Join one of our [community or developer meeting](https://keptn.sh/community/meetings/).

Reviewer: ```suggestion
4. Join one of our [community or developer meetings](https://keptn.sh/community/meetings/).
```

> 5. Try contributing to Keptn. Some project ideas include references to newcomer friendly issues, you can also find links to the generic contributing guide [here](https://keptn.sh/community/contributing/)

Reviewer: ```suggestion
5. Try contributing to Keptn. Some project ideas include references to newcomer friendly issues. You can also find links to the generic contributing guide [here](https://keptn.sh/community/contributing/)
```

6. Work with the potential mentors during the application phase.
   Do not hesitate to ask any questions in the chat.
   It might be possible to also organize special sessions to discuss the project ideas.

> 7. Make sure to submit the final application in time!

Author: Will address in a follow-up by adding a new "Key Dates" section

### Joining as mentor

We welcome contributors to join the mentoring teams.
Each team, if a project is accepted,

> will consist of 2 or more mentors and one student selected by the mentors 

Reviewer: ```suggestion
will consist of 2 or more mentors and one participant selected by the mentors 
```
I think that non-students qualify this year.

and GSoC admins during the application reviews.

**Expectations**. Mentors are expected to be experienced in the project idea domain
and to be willing to mentor students.

> We also expect mentors to be able to dedicate several hour per week during the community bonding and coding phases.

Reviewer: ```suggestion
We expect mentors to be able to dedicate several hour per week during the community bonding and coding phases.
```

References:

- [GSoC Mentor Guide](https://google.github.io/gsocguides/mentor/)
- [Mentor responsibilities](https://developers.google.com/open-source/gsoc/help/responsibilities#mentor_responsibilities)
- [GSoC Timeline](https://summerofcode.withgoogle.com/how-it-works#timeline)

### Proposing project ideas

> To add a new project, submit a pull requests against [this page](./2022/project-ideas.md).

Reviewer: ```suggestion
To add a new project idea, submit a pull requests against [this page](./2022/project-ideas.md).
```
